gnosis/safe-react
# fix: init Safe Apps providers per `chainId`

## What it solves
Resolves #3497

## How this PR fixes it
We were using a Safe Apps Web3 provider that didn't update with the chain info, meaning the RPC remained the same.

The provider is not instantiated depending on the changing RPC URL via a `useMemo` depedancy.

## How to test it
1. Open the Safe (on prod. CGW) on Optimisn and open the following Safe App: https://kwenta.io/.
2. Observe in the network tab the requests to the Optimism RPC `https://mainnet.optimism.io`.
3. Switch the Safe to Rinkeby and open this Safe App: https://rimeissner.dev/transaction-templates. Using https://raw.githubusercontent.com/rmeissner/transaction-templates/main/templates/compound.json with any amount, press "build".
5. Observe that it has requested the Rinkeby Infura RPC, not the Optimism one.
